packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -407,3 +434,287 @@ abstract class InlineSpan extends DiagnosticableTree {
+  ///
+  /// Must be null if [color] is set to a non-null value.
+  final ui.Paint? foreground;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.backgroundColor]
+  /// if set to non-null.
+  ///
+  /// Must be null if [background] is set to a non-null value.
+  final ui.Color? backgroundColor;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.background] if
+  /// set to non-null.
+  ///
+  /// Must be null if [backgroundColor] is set to a non-null value.
+  final ui.Paint? background;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.shadows] if set
+  /// to non-null.
+  final List<ui.Shadow>? shadows;
+
+  /// An attribute which overwrites the [TextDecoration.underline] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies underline and setting it to false
+  /// disables underline.
+  final bool? underline;
+
+  /// An attribute which overwrites the [TextDecoration.overline] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies overline and setting it to false
+  /// disables overline.
+  final bool? overline;
+
+  /// An attribute which overwrites the [TextDecoration.lineThrough] aspect of
+  /// [TextStyle.decoration] on an [InlineSpan], if set to non-null.
+  ///
+  /// Setting this value to true applies lineThrough and setting it to false
+  /// disables lineThrough.
+  final bool? lineThrough;
+
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationColor]
+  /// if set to non-null.
+  final ui.Color? decorationColor;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationStyle]
+  /// if set to non-null.
+  final ui.TextDecorationStyle? decorationStyle;
+  /// An attribute which overwrites an [InlineSpan]'s [TextStyle.decorationThickness]
+  /// if set to non-null.
+  final double? decorationThickness;
+
+  /// An attribute which overwrites [TextSpan.recognizer] if set to non-null.
+  ///
+  /// When a recognizer is specified, the [InlineSpan.updateAttributesAtOffset] method
+  /// replaces existing recognizers (if any) within the given range with the
+  /// specified recognizer.
+  ///
+  /// When this value is set to [remove], the [InlineSpan.updateAttributesAtOffset]
+  /// method sets [TextSpan.recognizer] to null in the given range of the returned
+  /// [TextSpan].
+  final GestureRecognizer? recognizer;
+
+  /// An attribute which overwrites [TextSpan.mouseCursor] if set to non-null.
+  final MouseCursor? mouseCursor;
+
+  /// An attribute which overwrites [TextSpan.onEnter] if set to non-null.
+  ///
+  /// [TextSpan.onEnter] can not be set from a non-null value to null using this
+  /// attribute. Considering setting [TextSpan.onEnter] to a function that does
+  /// nothing if you want to unset the callback from a [TextSpan].
+  final PointerEnterEventListener? onEnter;
```

Comment:
That default value could also work for these, reducing the number of remove options from 3 (unremovable, removable with `remove`, removable by setting to a no-op) to 2 (unremovable, use null to indicate remove).